tests/testthat/test-renderer2-widerect.R

```diff
@@ -1,454 +1,428 @@
-remDr$sendKeysToActiveElement(list(key="enter"))
-Sys.sleep(1)
-
-test_that("typing into selectize widget changes year to 1962", {
-  current.year <- getYear()
-  expect_identical(current.year, "1962")
-})
-
-s.div$clickElement()
-remDr$sendKeysToActiveElement(list(key="down_arrow"))
-remDr$sendKeysToActiveElement(list(key="enter"))
-Sys.sleep(1)
-
-test_that("down arrow key changes year to 1963", {
-  current.year <- getYear()
-  expect_identical(current.year, "1963")
-})
-
-getCountries <- function(){
-  country.labels <- getNodeSet(getHTML(), '//g[@class="geom8_text_ts"]//text')
-  sapply(country.labels, xmlValue)
-}
-
-test_that("initial countries same as first", {
-  country.vec <- getCountries()
-  expect_identical(sort(country.vec), sort(wb.facets$first$country))
-})
-
-s.tr <- remDr$findElement("class name", "country_variable_selector_widget")
-s.div <- s.tr$findChildElement("class name", "selectize-input")
-s.div$clickElement()
-remDr$sendKeysToActiveElement(list("Afg"))
-remDr$sendKeysToActiveElement(list(key="enter"))
-Sys.sleep(1)
-
-test_that("Afg autocompletes to Afghanistan", {
-  country.vec <- getCountries()
-  expected.countries <- c("United States", "Vietnam", "Afghanistan")
-  expect_identical(sort(country.vec), sort(expected.countries))
-})
-
-div.list <- s.tr$findChildElements("class name", "item")
-names(div.list) <- sapply(div.list, function(e)e$getElementText()[[1]])
-afg.div <- div.list[["Afghanistan"]]
-# clickElement has some really weird behavior, repeating it several times 
-# focuses different things and I can't reliably get it to actually focus on
-# the US element that the test was before.
-# This is kinda a hack that causes it to backspace the last element in the list
-afg.div$clickElement()
-remDr$sendKeysToActiveElement(list(key="backspace"))
-Sys.sleep(1)
-
-test_that("backspace removes Afghanistan from selected countries", {
-  country.vec <- getCountries()
-  expected.countries <- c("United States", "Vietnam")
-  expect_identical(sort(country.vec), sort(expected.countries))
-})
-
-getWidth <- function(){
-  node.set <-
-    getNodeSet(getHTML(), '//g[@class="geom10_bar_bar"]//rect[@id="Vietnam"]')
-  expect_equal(length(node.set), 1)
-  alist <- xmlAttrs(node.set[[1]])
-  alist[["width"]]
-}
-
-test_that("middle of transition != after when duration=2000", {
-  clickID("year1960")
-  Sys.sleep(1)
-  before.width <- getWidth()
-  clickID("year2010")
-  during.width <- getWidth()
-  Sys.sleep(0.1)
-  after.width <- getWidth()
-  rbind(before=before.width,
-        during=during.width,
-        after=after.width)
-  expect_true(during.width != after.width)
-})
-
-e <- remDr$findElement("id", "plot_duration_ms_year")
-e$clickElement()
-e$clearElement()
-e$sendKeysToElement(list("0", key="enter"))
-Sys.sleep(1)
-
-test_that("middle of transition == after when duration=0", {
-  clickID("year1960")
-  Sys.sleep(1)
-  before.width <- getWidth()
-  clickID("year2010")
-  during.width <- getWidth()
-  Sys.sleep(0.1)

```

Comment:
I believe this is the only test of smooth transitions so I don't think we should delete it.
can you please undo deletion and try to adjust the constants in order to get it to pass?

Reply:
@tdhock Added back the deleted test. The second condition checked in the test fails on chromote: 
expect_true(during.width == after.width)

The width during the transition is not equal to width after when plot duration year = 0. I am not sure if this is expected behavior, is the transition not expected to happen in that case ?

Reply:
plot duration year = 0 means to not use smooth transitoins, instead transition instantly, for updates to the year variable.
positive values of duration mean to use that many milliseconds for the duration of the smooth transition.
so there should be one test with 0 and one test with a large positive value. and the during should be either same as end (for duration 0 / instant transition) or some intermediate value (for positive duration / smooth transition)

Reply:
Thanks for the explanation , the test failed earlier because I was not setting plot duration year = 0 correctly. I was able to fix that by sending a JavaScript event after setting the value to 0. The test passes now and there are no smooth transitions when plot duration year is set to 0